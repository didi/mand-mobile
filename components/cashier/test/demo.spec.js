import Demo0 from './cases/demo0'
import Demo1 from './cases/demo1'
import Demo2 from './cases/demo2'
import Demo3 from './cases/demo3'
import Demo4 from './cases/demo4'
import {renderToString} from '@vue/server-test-utils'
import {mount} from '@vue/test-utils'

describe('Cashier', () => {
  test(`Baisc-channel`, () => {
    expect(renderToString(Demo0)).toMatchSnapshot()
  })
  test(`Baisc-captcha`, done => {
    const wrapper = mount(Demo1)
    wrapper.vm.$nextTick(() => {
      expect(wrapper.vm.$el.innerHTML).toMatchSnapshot()
      done()
    })
  })
  test(`Baisc-success`, done => {
    const wrapper = mount(Demo2)
    wrapper.vm.$nextTick(() => {
      expect(wrapper.vm.$el.innerHTML).toMatchSnapshot()
      done()
    })
  })
  test(`Baisc-fail`, done => {
    const wrapper = mount(Demo3)
    wrapper.vm.$nextTick(() => {
      expect(wrapper.vm.$el.innerHTML).toMatchSnapshot()
      done()
    })
  })
  test(`Using slots and other configurations`, () => {
    expect(renderToString(Demo4)).toMatchSnapshot()
  })
})
