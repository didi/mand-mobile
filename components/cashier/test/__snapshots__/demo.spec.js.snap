// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cashier - Demo Baisc-captcha 1`] = `
<div class="md-cashier">
  <div class="md-popup inner-popup with-mask bottom" style="">
    <div class="md-popup-mask" style="" name="md-mask-fade"></div>
    <div class="md-popup-box md-slide-up" style="" name="md-slide-up">
      <div class="md-popup-title-bar title-align-center">
        <!---->
        <div class="title-bar-title">
          <p class="title">支付</p>
          <!---->
        </div>
        <!---->
        <div class="title-bar-right md-popup-close"><i class="md-icon icon-font md-icon-close close lg"></i></div>
      </div>
      <div class="md-cashier-container">
        <div class="md-cashier-block md-cashier-captcha">
          <div class="md-captcha">
            <div class="md-captcha-content">
              <!---->
              <div class="md-captcha-message">
                <div>Verification code sent to 156 **** 8965</div>
              </div>
            </div>
            <div class="md-codebox-wrapper">
              <div class="md-codebox is-justify"><span class="md-codebox-box"><!----> <!----></span><span class="md-codebox-box"><!----> <!----></span><span class="md-codebox-box"><!----> <!----></span><span class="md-codebox-box"><!----> <!----></span></div>
              <footer class="md-captcha-footer">
                <div class="md-captcha-brief">The latest verification code is still valid</div> <button class="md-captcha-btn">Send Verification code</button>
              </footer>
              <form action="" style="display: none;"><input type="tel" maxlength="4" class="md-codebox-input"></form>
              <div class="md-number-keyboard md-codebox-keyboard in-view">
                <!---->
                <div class="md-number-keyboard-container simple">
                  <div class="keyboard-number">
                    <ul class="keyboard-number-list">
                      <li class="keyboard-number-item"><span>1</span></li>
                      <li class="keyboard-number-item"><span>2</span></li>
                      <li class="keyboard-number-item"><span>3</span></li>
                      <li class="keyboard-number-item"><span>4</span></li>
                      <li class="keyboard-number-item"><span>5</span></li>
                      <li class="keyboard-number-item"><span>6</span></li>
                      <li class="keyboard-number-item"><span>7</span></li>
                      <li class="keyboard-number-item"><span>8</span></li>
                      <li class="keyboard-number-item"><span>9</span></li>
                      <li class="keyboard-number-item no-bg"></li>
                      <li class="keyboard-number-item"><span>0</span></li>
                      <li class="keyboard-number-item no-bg delete"><span></span></li>
                    </ul>
                  </div>
                  <!---->
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Cashier - Demo Baisc-channel 1`] = `
<div class="md-example-child md-example-child-cashier">
  <div class="md-cashier">
    <div class="md-popup inner-popup with-mask bottom" style="display:none;">
      <div name="md-mask-fade" class="md-popup-mask" style="display:none;"></div>
      <div name="md-slide-up" class="md-popup-box md-slide-up" style="display:none;">
        <div class="md-popup-title-bar title-align-center">
          <!---->
          <div class="title-bar-title">
            <p class="title">支付</p>
            <!---->
          </div>
          <!---->
          <div class="title-bar-right md-popup-close"><i class="md-icon icon-font md-icon-close close lg" style="color:;"></i></div>
        </div>
        <div class="md-cashier-container">
          <div class="md-cashier-block md-cashier-choose">
            <div class="md-cashier-channel">
              <div class="choose-text">
                <p class="choose-title">支付金额(元)</p>
                <p class="choose-number">100.00</p>
                <p class="choose-describe">关于支付金额的特殊说明</p>
              </div>
              <div class="choose-channel">
                <div class="choose-channel-list">
                  <div class="md-cashier-channel-item default">
                    <div class="item-icon cashier-icon-1"><i class="md-icon icon-font md-icon-cashier-icon-1 cashier-icon-1 lg" style="color:;"></i></div>
                    <div class="item-label">
                      <p class="title"><span>招商银行(0056)</span>
                        <!---->
                      </p>
                      <!---->
                    </div>
                    <div class="item-check-icon"><i class="md-icon icon-font md-icon-checked checked md" style="color:;"></i></div>
                  </div>
                </div>
                <div class="choose-channel-more">更多支付方式</div>
              </div>
              <div class="md-cashier-block-btn"><button type="button" class="md-button md-cashier-pay-button primary active block">
                  <div class="md-button-inner">
                    <!---->
                    <div class="md-button-content">确定支付</div>
                  </div>
                </button></div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Cashier - Demo Baisc-fail 1`] = `
<div class="md-cashier">
  <div class="md-popup inner-popup with-mask bottom" style="">
    <div class="md-popup-mask" style="" name="md-mask-fade"></div>
    <div class="md-popup-box md-slide-up" style="" name="md-slide-up">
      <div class="md-popup-title-bar title-align-center">
        <!---->
        <div class="title-bar-title">
          <p class="title">支付</p>
          <!---->
        </div>
        <!---->
        <div class="title-bar-right md-popup-close"><i class="md-icon icon-font md-icon-close close lg"></i></div>
      </div>
      <div class="md-cashier-container">
        <div class="md-cashier-block md-cashier-fail">
          <div class="md-cashier-block-icon"><svg class="md-icon icon-svg md-icon-warn-color md">
              <use xlink:href="#warn-color"></use>
            </svg></div>
          <div class="md-cashier-block-text">支付失败，请稍后重试</div>
          <div class="md-cashier-block-btn"><button type="button" class="md-button primary active block">好的</button></div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Cashier - Demo Baisc-success 1`] = `
<div class="md-cashier">
  <div class="md-popup inner-popup with-mask bottom" style="">
    <div class="md-popup-mask" style="" name="md-mask-fade"></div>
    <div class="md-popup-box md-slide-up" style="" name="md-slide-up">
      <div class="md-popup-title-bar title-align-center">
        <!---->
        <div class="title-bar-title">
          <p class="title">支付</p>
          <!---->
        </div>
        <!---->
        <div class="title-bar-right md-popup-close"><i class="md-icon icon-font md-icon-close close lg"></i></div>
      </div>
      <div class="md-cashier-container">
        <div class="md-cashier-block md-cashier-success">
          <div class="md-cashier-block-icon">
            <div class="md-activity-indicator-rolling-success">
              <div class="md-activity-indicator-rolling" radius="35">
                <div class="rolling-container"><svg viewBox="0 0 78.75 78.75" preserveAspectRatio="xMidYMid" class="md-activity-indicator-svg rolling" style="width: 70px; height: 70px; transform: rotateZ(0deg);">
                    <circle fill="none" stroke="transparent" stroke-width="4.375" cx="39.375" cy="39.375" r="35"></circle>
                    <g name="circle">
                      <circle cx="39.375" cy="39.375" fill="#FFF6F1" stroke="none" r="35" class="success"></circle>
                    </g>
                  </svg>
                  <div class="content"> <svg viewBox="0 0 100 100" preserveAspectRatio="xMidYMid" class="right" style="transform: translate(-50%, -50%) scale(1);">
                      <g>
                        <line x1="32" y1="47" x2="45" y2="62" stroke-dasharray="20" style="stroke-width: 4.375; stroke: #2F86F6;"></line>
                        <line x1="42" y1="62" x2="68.4" y2="40" stroke-dasharray="35" style="stroke-width: 4.375; stroke: #2F86F6;"></line>
                      </g>
                    </svg></div>
                </div>
              </div>
            </div>
          </div>
          <div class="md-cashier-block-text">支付成功</div>
          <div class="md-cashier-block-btn"><button type="button" class="md-button primary active block">好的</button></div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Cashier - Demo Using slots and other configurations 1`] = `
<div class="md-example-child md-example-child-cashier">
  <div class="md-cashier">
    <div class="md-popup inner-popup with-mask bottom" style="display:none;">
      <div name="md-mask-fade" class="md-popup-mask" style="display:none;"></div>
      <div name="md-slide-up" class="md-popup-box md-slide-up" style="display:none;">
        <div class="md-popup-title-bar title-align-center">
          <!---->
          <div class="title-bar-title">
            <p class="title">支付</p>
            <!---->
          </div>
          <!---->
          <div class="title-bar-right md-popup-close"><i class="md-icon icon-font md-icon-close close lg" style="color:;"></i></div>
        </div>
        <div class="md-cashier-container">
          <div>
            <div class="md-notice-bar warning">
              <div class="md-notice-bar-left"><i class="md-icon icon-font md-notice-icon md-icon-warn warn md" style="color:;"></i></div>
              <div class="md-notice-bar-content">
                <div>
                  该银行3:00-12:00系统维护，请更换其他银行卡
                </div>
              </div>
              <div class="md-notice-bar-right"><i class="md-icon icon-font md-notice-icon md-notice-icon-right md-icon-close close md" style="color:;"></i></div>
            </div>
          </div>
          <div class="md-cashier-block md-cashier-choose">
            <div class="md-cashier-channel">
              <div class="choose-text">
                <p class="choose-title">支付金额(元)</p>
                <p class="choose-number">100.00</p>
                <p class="choose-describe">关于支付金额的特殊说明</p>
              </div>
              <div class="choose-channel">
                <div class="choose-channel-list">
                  <div class="md-cashier-channel-item default">
                    <div class="item-icon cashier-icon-1"><i class="md-icon icon-font md-icon-cashier-icon-1 cashier-icon-1 lg" style="color:;"></i></div>
                    <div class="item-label">
                      <p class="title"><span>XX银行(1234)</span> <span class="title-active">更换</span></p>
                      <p class="desc">当前银行维护中</p>
                    </div>
                    <div class="item-check-icon"><i class="md-icon icon-font md-icon-check-disabled check-disabled md" style="color:;"></i></div>
                  </div>
                </div>
                <!---->
              </div>
              <div class="md-cashier-block-btn"><button type="button" class="md-button md-cashier-pay-button primary active block">
                  <div class="md-button-inner">
                    <!---->
                    <div class="md-button-content">确定支付</div>
                  </div>
                </button></div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
